src/signals/incident/components/form/MapSelectors/Asset/Selector/WfsLayer/StatusLayer/StatusLayer.tsx

```diff
@@ -5,51 +5,42 @@ import './style.css'
 
 import type { FC } from 'react'
 import type {
-  CheckedFeatureType,
   Feature,
-  ReportedFeatureType,
+  FeatureStatusType,
 } from 'signals/incident/components/form/MapSelectors/types'
 import { Marker } from '@amsterdam/arm-core'
 import { featureToCoordinates } from 'shared/services/map-location'
 import type { Geometrie } from 'types/incident'
-import { getIsChecked, getIsReported } from './utils'
+import { getFeatureStatusType } from './utils'
 
 const STATUS_CLASS_MODIFIER = 'marker-status'
 
 export interface StatusLayerProps {
   statusFeatures: Feature[]
-  reportedFeatureType: ReportedFeatureType
-  checkedFeatureType?: CheckedFeatureType
+  featureStatusTypes: FeatureStatusType[]
 }
 
 const StatusLayer: FC<StatusLayerProps> = ({
   statusFeatures,
-  reportedFeatureType,
-  checkedFeatureType,
+  featureStatusTypes,
 }) => {
   const getMarker = (feat: any, index: number) => {
     const feature = feat as Feature
     const latLng = featureToCoordinates(feature?.geometry as Geometrie)
 
-    const isReported = getIsReported(feature, reportedFeatureType)
-    const isChecked = getIsChecked(feature, checkedFeatureType)
+    const featureStatusType = getFeatureStatusType(feature, featureStatusTypes)
+    if (!featureStatusType) {
+      return null
+    }
```

Comment:
This situation shouldn't happen because there should always be a featureStatusType. I don't know how the typing issues can be fixed in a different way. Do you have a suggestion?

Reply:
Yes, you can turn the function into a generic